packages/cozy-konnector-libs/src/libs/saveFiles.js

```diff
@@ -188,24 +213,102 @@ const removeFile = async function(file) {
   await cozy.files.destroyById(file._id)
 }
 
+async function getFileIfExists(entry, options) {
+  const filePrimaryKeys = options.filePrimaryKeys
+  if (!filePrimaryKeys) {
+    log(
+      'warn',
+      `saveFiles: no deduplication key is defined, file deduplication will be based on file path`
+    )
+  }
+
+  const slug = manifest.data.slug
+  if (!slug) {
+    log(
+      'warn',
+      `saveFiles: no slug is defined for the current connector, file deduplication will be based on file path`
+    )
+  }
+
+  const sourceAccountIdentifier = get(
+    options,
+    'sourceAccountOptions.sourceAccountIdentifier'
+  )
+  if (!sourceAccountIdentifier) {
+    log(
+      'warn',
+      `saveFiles: no sourceAccountIdentifier is defined in options, file deduplication will be based on file path`
+    )
+  }
+
+  if (!filePrimaryKeys || !slug || !sourceAccountIdentifier) {
+    try {
+      const result = await cozy.files.statByPath(
+        getFilePath({ entry, options })
+      )
+      return result
+    } catch (err) {
+      log('debug', err.message)
+      return false
+    }
+  } else {
+    const index = await cozy.data.defineIndex('io.cozy.files', [
+      'metadata.filePrimaryKeys',
+      'trashed',
+      'cozyMetadata.sourceAccountIdentifier',
+      'cozyMetadata.createdByApp'
+    ])
+    const files = await queryAll(
+      'io.cozy.files',
+      {
+        metadata: {
+          filePrimaryKeys: calculateFileKey(entry, filePrimaryKeys)
+        },
+        trashed: false,
+        cozyMetadata: {
+          sourceAccountIdentifier: sourceAccountIdentifier,
+          createdByApp: slug
+        }
+      },
+      index
+    )
+    if (files && files[0]) {
+      if (files.length > 1) {
+        log(
+          'warn',
+          `Found ${files.length} files corresponding to ${calculateFileKey(
+            entry,
+            filePrimaryKeys
+          )}`
+        )
+      }
+      return files[0]
+    } else {
+      try {
+        const result = await cozy.files.statByPath(
+          getFilePath({ entry, options })
+        )
+        return result
+      } catch (err) {
+        log('debug', err.message)
+        return false
+      }
```

Comment:
The try/catch can be extracted to a function `getFileByPathIfExists` since it is the same as above.

Reply:
I did a small refactor and used another name bit should be better